aten/src/ATen/native/cuda/LinearAlgebra.cu

```diff
@@ -126,7 +127,7 @@ Tensor& addmm_out_cuda_impl(Tensor& result, const Tensor& self, const Tensor& ma
     if (beta.toComplexDouble() == 0.) {
       return result.zero_();
     }
-    return at::native::mul_out(result, self, at::native::scalar_tensor(beta, at::device(at::kCPU).dtype(self.scalar_type())));
+    return at::cuda::mul_out(result, self, at::native::scalar_tensor(beta, at::device(at::kCPU).dtype(self.scalar_type())));
```

Comment:
Technically, I think `at::cpu::mul_out` is more in-line with the original behavior than `at::cuda::mul_out`. They both eventually call the structured kernel implementation, which uses DispatchStub to dispatch appropriately to cpu/cuda.

The only difference is that `at::cuda::mul_out` sets up a cuda guard before calling into the kernel. At this point in the code we're already inside a cuda kernel, so I think that the extra guard setup might be redundant.

To me it seems more in the right spirit to call the cuda variant than the cpu variant, since:
- Calling the cpu variant here would look confusing to people who don't know what `at::cpu::mul_out` actually does
- calling the cuda variant protects against any future divergences in the cuda/cpu implementations.

Reply:
Clearly this means we should have a `cuda_unchecked` namespace lololol

Reply:
(Switching this to cuda is fine by me)

Reply:
@robieta It suuuuure would be nice if we had a comprehensive set of microbenchmarks that would notice when an extra device guard was called, as it is in this case :)